libclc/utils/libclc-remangler/CMakeLists.txt

```diff
@@ -25,6 +25,24 @@ clang_target_link_libraries(libclc-remangler
   LLVMOption
   )
 
+if(CMAKE_CROSSCOMPILING)
+  if(LLVM_NATIVE_TOOL_DIR AND NOT LIBCLC_REMANGLER_PATH)
+    if(EXISTS "${LLVM_NATIVE_TOOL_DIR}/libclc-remangler${LLVM_HOST_EXECUTABLE_SUFFIX}")
+      set(LLVM_CONFIG_PATH "${LLVM_NATIVE_TOOL_DIR}/libclc-remangler${LLVM_HOST_EXECUTABLE_SUFFIX}")
+    endif()
+  endif()
+
+  if(NOT LIBCLC_REMANGLER_PATH)
+    build_native_tool(libclc-remangler LIBCLC_REMANGLER_PATH)
+    set(LIBCLC_REMANGLER_PATH "${LIBCLC_REMANGLER_PATH}" CACHE STRING "")
+
+    add_custom_target(NativeLibCLCRemangler DEPENDS ${LIBCLC_REMANGLER_PATH})
+    add_dependencies(libclc-remangler NativeLibCLCRemangler)
+  endif()
+else()
+  set(LIBCLC_REMANGLER_PATH "${CMAKE_CURRENT_BINARY_DIR}/bin/libclc-remangler" CACHE STRING "")
```

Comment:
I'm not sure relying on the path like this is the best idea. Perhaps we can get the path to the binary using the `libclc-remangler` target itself? Does `$<TARGET_PROPERTY:libclc-remangler,TARGET_FILE>` work?

Reply:
Referencing again 😅:

> Same thing as my previous comment above, but yes:
> 
> > Yes it should've but I have decided to not do all the cross compiling logic manually, instead using `setup_host_tool`.